src/common/state/redis.c

```diff
@@ -380,32 +578,32 @@ void task_log_redis_callback(redisAsyncContext *c, void *r, void *privdata) {
     return;
   }
   /* Otherwise, parse the task and call the callback. */
-  task_log_subscribe_data *data = callback_data->data;
+  task_table_subscribe_data *data = callback_data->data;
 
-  task_instance *instance = malloc(reply->element[2]->len);
-  memcpy(instance, reply->element[2]->str, reply->element[2]->len);
+  task *task = malloc(payload->len);
+  memcpy(task, payload->str, payload->len);
   if (data->subscribe_callback) {
-    data->subscribe_callback(instance, data->subscribe_context);
+    data->subscribe_callback(task, data->subscribe_context);
   }
-  task_instance_free(instance);
+  free_task(task);
 }
 
-void redis_task_log_subscribe(table_callback_data *callback_data) {
+void redis_task_table_subscribe(table_callback_data *callback_data) {
   db_handle *db = callback_data->db_handle;
-  task_log_subscribe_data *data = callback_data->data;
+  task_table_subscribe_data *data = callback_data->data;
 
-  if (memcmp(&data->node.id[0], &NIL_ID.id[0], UNIQUE_ID_SIZE) == 0) {
-    redisAsyncCommand(db->sub_context, task_log_redis_callback,
-                      (void *) callback_data->timer_id,
-                      "PSUBSCRIBE task_log:*:%d", data->state_filter);
+  if (IS_NIL_ID(data->node)) {
```

Comment:
I wonder if these two cases are really so similar that they should be put in the same method.

In particular, in the case when we subscribe to changes to all tasks, are we sure we only want to subscribe to all future tasks? We may want to get all past changes as well (e.g., the very first global scheduler needs to get anything that is already in Redis).